fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreFormatVersionTest.java

```diff
@@ -40,23 +44,25 @@ public class FDBRecordStoreFormatVersionTest extends FDBRecordStoreTestBase {
 
     @Test
     public void testFormatVersionUpgrade() {
+        final List<FormatVersion> sortedVersions = Arrays.stream(FormatVersion.values()).sorted().collect(Collectors.toList());
+        FormatVersion penultimateVersion = sortedVersions.get(sortedVersions.size() - 1);
         try (FDBRecordContext context = openContext()) {
             recordStore = getStoreBuilder(context, simpleMetaData(NO_HOOK))
-                    .setFormatVersion(FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION - 1)
+                    .setFormatVersion(penultimateVersion)
```

Comment:
I thought about introducing a `FormatVersion.previous()`, but it would only be used for internal tests, and you can't change the numbers associated with the enums, so it seemed best to keep it simple.